### PR TITLE
PRO-2079 PLR Staking Changes

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-2079/plr-staking-changes

## Description

- Implemented PLR Staking on Polygon

## How Has This Been Tested?

- Tested on Etherspot BUIDLer.

## Screenshots (if appropriate):
Validation Message:
Screenshot 2023-12-20 at 11 45 01 AM

Steps:

https://github.com/etherspot/etherspot-react-transaction-buidler/assets/82652040/ccd27d3b-d4e0-4107-badf-0f4ab11e4d88

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/TransactionBlock/PlrStakingV2TransactionBlock.tsx [hunk 1]
```diff
@@ -477,59 +466,62 @@ const PlrStakingV2TransactionBlock = ({
...
+
+              try {
+                const { items: balances } = await sdk.getAccountBalances({
+                  account: address,
+                  tokens: tokensForBalance,
+                  chainId: chain.chainId,
+                });
+
+                if (!shouldUpdate) return;
+
```
Comment (IAmKio) [inline src/components/TransactionBlock/PlrStakingV2TransactionBlock.tsx:492]: This makes me want to cry 😢

**Review by IAmKio — APPROVED** [review]
Approving this, much of the changed files have been linted and actually it appears not too much changed in terms of the actual code to complete the feature.

File: src/hooks/useEtherspot.tsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
-import React, { useContext } from 'react';
+import { useContext } from 'react';
 
 import { EtherspotContext } from '../contexts';
 
-const useEtherspot = () => {
+const useEtherspot = (): any => {
```
Comment (poocart) [inline src/hooks/useEtherspot.tsx:5]: please avoid `any`

Comment (poocart) [reply src/hooks/useEtherspot.tsx:5]: there's already a type for this within context provider

**Pull request merged**